### PR TITLE
Implement Bookshare search and search results pages

### PR DESCRIPTION
**Pull request by klown**
DRAFT

- [x] Add “Add from Bookshare” button to “Library (My Readings)” page
- [x] Search dialog page – “Add from Bookshare Step 1”
- [x] List search results page – “Add from Bookshare Step 2”.
- [x] Need to be aware of books already imported and list them “greyed out”
  - List titles with text "Already imported"
- [x] Need footer with  “previous”, “next”, and page number links since a page shows only N results at a time.
- [ ] Implement sorting of search results.  Sort by relevance, title, author, ISBN, Copyright data, or Date Added (Boris: defer this).
- [x] Retrieve link to thumbnail from search results for display
- [x] Handle synopsis

JIRA: https://castudl.atlassian.net/browse/CSL-1412

### CODE REVIEW

File: src/library/templates/library/library_bookshare_import.html [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                var elementToShow = document.getElementById(`long_${id}`);
+                var elementToHide = document.getElementById(`short_${id}`);
+            }
+            else if (action === "less_") {
+                var elementToHide = document.getElementById(`long_${id}`);
+                var elementToShow = document.getElementById(`short_${id}`);
+            }
+            elementToHide.style.display = 'none';
+            elementToShow.style.display = 'block';
+            event.stopPropagation();
```
Comment (mbrambilla) [inline src/library/templates/library/library_bookshare_import.html:72]: Most likely want click event to be able to propagate - it is possible that something like event logging could be hooked into these controls in the future. Event logging uses delegated events, `event.stopPropagation()` would block this functionality.

File: src/library/templates/library/library_bookshare_import.html [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                            <li>Author: {{ author }}</li>
+                            {% empty %}
+                            <li>No author</li>
+                            {% endfor %}
+                            <li>ISBN: {{ title.isbn13 }}</li>
+                            <li><img src="{{ title.clusive.thumbnail }}" alt="Cover of {{ title.title }}"</img></li>
+                            {% if title.clusive.more %}
+                            <li id="short_{{ title.bookshareId }}">
+                                Description: {{ title.clusive.synopsis }} &hellip;
+                                <button id="more_{{title.bookshareId}}" class="link-btn more-or-less">See more</button>
```
Comment (mbrambilla) [inline src/library/templates/library/library_bookshare_import.html:35]: Needs some disambiguation - suggest adding book title to button, , visually hidden but screen readers can see it.
Something like
`<button id="more_{{title.bookshareId}}" class="link-btn more-or-less">See more <span class="sr-only">about {{ title.title }}</span></button>`

File: src/library/templates/library/library_bookshare_import.html [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                            </li>
+                            <li id="long_{{ title.bookshareId }}" style="display: none">
+                                Description: {{ title.synopsis }}
+                                <button id="less_{{title.bookshareId}}" class="link-btn more-or-less">See less</button>
```
Comment (mbrambilla) [inline src/library/templates/library/library_bookshare_import.html:39]: Similar to more button - should try to reduce duplicate named controls.

File: src/library/templates/library/library_bookshare_search_results.html [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                    var elementToShow = document.getElementById(`long_${id}`);
+                    var elementToHide = document.getElementById(`short_${id}`);
+                }
+                else if (action === "less_") {
+                    var elementToHide = document.getElementById(`long_${id}`);
+                    var elementToShow = document.getElementById(`short_${id}`);
+                }
+                elementToHide.style.display = 'none';
+                elementToShow.style.display = 'block';
+                event.stopPropagation();
```
Comment (mbrambilla) [inline src/library/templates/library/library_bookshare_search_results.html:137]: Should be safe to remove stop propagation.

File: src/library/templates/library/library_bookshare_search_results.html [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+                                                {% empty %}
+                                                <div>No author listed</div>
+                                                {% endfor %}
+                                                <div>ISBN: {{ book.isbn13 }} ({{ book.clusive.epub }})</div>
+                                            </div>
+                                            <div class="card-text">
+                                                {% if book.clusive.more %}
+                                                <p id="short_{{ book.bookshareId }}">
+                                                    {{ book.clusive.synopsis }} &hellip;
+                                                    <button id="more_{{ book.bookshareId }}" class="link-btn more-or-less">See more</button>
```
Comment (mbrambilla) [inline src/library/templates/library/library_bookshare_search_results.html:64]: Another place for trying to reduce duplicate named controls

File: src/library/templates/library/library_bookshare_search_results.html [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+                                                </p>
+                                                <p id="long_{{ book.bookshareId }}" style="display: none">
+                                                    {{ book.synopsis }}
+                                                    <button id="less_{{ book.bookshareId }}" class="link-btn more-or-less">See less</button>
```
Comment (mbrambilla) [inline src/library/templates/library/library_bookshare_search_results.html:68]: Another place for trying to reduce duplicate named controls

File: src/library/templates/library/library_bookshare_import.html (hunk 1 above)
Comment (klown) [reply src/library/templates/library/library_bookshare_import.html]: Right.  I've deleted the line.

**Pull request merged**